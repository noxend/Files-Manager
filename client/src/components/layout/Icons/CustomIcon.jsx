import React from 'react';
import { ReactComponent as Docx } from '../../../assets/docx.svg';
import { ReactComponent as Pptx } from '../../../assets/pptx.svg';
import { ReactComponent as Doc } from '../../../assets/doc.svg';
import { ReactComponent as Psd } from '../../../assets/psd.svg';
import { ReactComponent as Ppt } from '../../../assets/ppt.svg';
import { ReactComponent as Jpg } from '../../../assets/jpg.svg';
import { ReactComponent as Pdf } from '../../../assets/pdf.svg';
import { ReactComponent as Png } from '../../../assets/png.svg';
import { ReactComponent as Txt } from '../../../assets/txt.svg';

const CustomIcon = props => {
  switch (props.type) {
    case 'docx':
      return <Docx {...props} />;
    case 'doc':
      return <Doc {...props} />;
    case 'psd':
      return <Psd {...props} />;
    case 'pptx':
      return <Pptx {...props} />;
    case 'ppt':
      return <Ppt {...props} />;
    case 'pdf':
      return <Pdf {...props} />;
    case 'jpg':
      return <Jpg {...props} />;
    case 'png':
      return <Png {...props} />;
    case 'txt':
      return <Txt {...props} />;
    default:
      return <Png {...props} />;
  }
};

export default CustomIcon;
