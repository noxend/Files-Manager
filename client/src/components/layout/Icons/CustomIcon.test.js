import React from 'react';
import renderer from 'react-test-renderer';

import CustomIcon from './CustomIcon';

describe('Custom Icon', () => {
  it('The Snapshot', () => {
    const tree = renderer.create(<CustomIcon type="docx" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
