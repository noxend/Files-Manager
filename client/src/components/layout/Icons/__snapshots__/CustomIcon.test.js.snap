// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Icon The Snapshot 1`] = `
<svg
  type="docx"
>
  docx.svg
</svg>
`;
